### PR TITLE
fix exitcode regression; improve testing; differentiate exit codes

### LINKED ISSUE #331: testify 0.9.0 discovery errors silently fail (exit 0)
This is a regression in 0.9.0, most likely due to #329

```python
$ tail -n999 tests/*.py
==> tests/__init__.py <==

==> tests/test1.py <==
import testify as T


class Test(T.TestCase):
    def test(self):
        print('test1')

==> tests/test2.py <==
import i_dont_exist
```

```
$ testify tests
test1
.Traceback (most recent call last):
  File "/tmp/testing/venv/local/lib/python2.7/site-packages/testify/test_discovery.py", line 97, in discover
    submod = __import__(module_name, fromlist=[str('__trash')])
  File "./tests/test2.py", line 1, in <module>
    import i_dont_exist
ImportError: No module named i_dont_exist
DISCOVERY FAILURE!
There was a problem importing one or more tests:

    Traceback (most recent call last):
      File "/tmp/testing/venv/local/lib/python2.7/site-packages/testify/test_discovery.py", line 97, in discover
        submod = __import__(module_name, fromlist=[str('__trash')])
      File "./tests/test2.py", line 1, in <module>
        import i_dont_exist
    ImportError: No module named i_dont_exist

PASSED.  1 test / 1 case: 1 passed, 0 failed.  (Total test time 0.00s)
$ echo $?
0
```

### PR DESCRIPTION
**Pull request by bukzor**
closes #331
closes internal ticket CDYM-420

I'm going to look at the test-raises-SystemExit case next.

### CODE REVIEW

File: testify/test_runner.py [hunk 1]
```diff
@@ -31,6 +30,10 @@
 __author__ = "Oliver Nicholas <bigo@yelp.com>"
 __testify = 1
 
+EX_OK = 0  # successful termination
+EX_DATAERR = 65  # EX_DATAERR -- The input data was incorrect in some way.
+EX_SOFTWARE = 70  # EX_SOFTWARE -- An internal software error has been detected.
```
Comment (asottile) [inline testify/test_runner.py]: let's give these semantic names, how about `EXIT_OK` or `EXIT_CODE_OK`

Comment (asottile) [inline testify/test_runner.py]: also no reason to repeat the constant name in the comment about the constant name, I think it's fairly obvious what you are referring to :)

Comment (asottile) [inline testify/test_runner.py]: also, maybe we should pick different numbers to be able to differentiate between unexpected traceback (I think exit code 1?) -- perhaps p3 though

File: testify/test_runner.py [hunk 2]
```diff
@@ -118,13 +118,8 @@ def discover_tests_testing():
...
 
-        try:
-            for test in discovered_tests:
-                yield test
-        except test_discovery.DiscoveryError as exc:
-            for reporter in self.test_reporters:
-                reporter.test_discovery_failure(exc)
-            sys.exit(1)
+        for test in discovered_tests:
+            yield test
```
Comment (asottile) [inline testify/test_runner.py]: can just be `return discovered_tests` ?

File: testify/test_program.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import imp
 
 import testify
+from testify import exit
```
Comment (asottile) [inline testify/test_program.py]: masks builtin `exit`, wev

**Comment by asottile** [comment]
:boat:

File: testify/test_runner.py (hunk 2 above)
Comment (bukzor) [inline testify/test_runner.py]: Looks like yes.

**Pull request merged**